libs/hdf-converters/src/converters-from-hdf/caat/reverse-caat-mapper.ts

```diff
@@ -0,0 +1,257 @@
+      : '';
+    return `${caveat}${this.fix(hdf.wraps.desc)}`;
+  }
+
+  createTestResultDescription(hdf: HDFControl): string {
+    const controlStatus = `${hdf.status}:\r\n\r\n`;
+    const description =
+      hdf.segments?.map((result) => {
+        const statusAndTest = `${result.status.toUpperCase()} -- Test: ${
+          result.code_desc
+        }\r\n`;
+        const message = result.message
+          ? `Message: ${result.message}\r\n\r\n`
+          : '\r\n';
+        return `${statusAndTest}${message}`;
+      }).join('') ?? '';
+    return `${controlStatus}${description}`;
+  }
+
+  createTestResult(hdf: HDFControl): string {
+    return hdf.status === 'Passed' ? 'Satisfied' : 'Other Than Satisfied';
+  }
+
+  createImpact(hdf: HDFControl): string {
+    const controlSeverity =
+      hdf.severity === 'medium' ? 'moderate' : hdf.severity;
+    return this.fix(hdf.wraps.impact === 0 ? 'none' : controlSeverity);
+  }
+
+  getRow(control: ContextualizedControl, filename: string): CAATRow[] {
+    const hdf = control.hdf;
+    const allRows: CAATRow[] = _.compact(
+      hdf
+        .canonized_nist(FromHDFToCAATMapper.NistCanonizationConfig)
+        .map((formattedNistTag) => {
+          // I don't think the canonized_nist function would return something that wasn't a nist tag, but this check was there originally quite probably for a reason, but I've not been able to find a sample that triggers this case myself
+          if (!formattedNistTag) {
+            // early exiting forces us to use the compact function to get rid of empty values in the array
+            return;
+          }
+
+          const row: CAATRow = {};
+          row['Control Number'] = formattedNistTag;
+          row['Finding Title'] = `Test ${this.fix(hdf.wraps.id)} - ${this.fix(
+            hdf.wraps.title
+          )}`;
+          if (hdf.start_time) {
+            row['Date Identified'] = FromHDFToCAATMapper.formatDate(
+              new Date(hdf.start_time),
+              '/'
+            );
+          }
+          row['Finding ID'] = `${filename} - Test ${this.fix(hdf.wraps.id)}`;
+          row['Finding Description'] = this.fix(hdf.wraps.title);
+          row['Weakness Description'] = this.createCaveat(hdf);
+          row['Control Weakness Type'] = 'Security';
+          row['Source'] = 'Self-Assessment';
+          row['Test Method'] = 'Test';
+          row['Test Objective'] = this.fix(
+            hdf.descriptions.check ?? hdf.wraps.tags.check
+          );
+          row['Test Result Description'] = this.fix(
+            this.createTestResultDescription(hdf)
+          );
+          row['Test Result'] = this.createTestResult(hdf);
+          row['Recommended Corrective Action(s)'] = this.fix(
+            hdf.descriptions.fix ?? hdf.wraps.tags.fix
+          );
+          row['Impact'] = this.createImpact(hdf);
+          return row;
+        })
+    );
+    return allRows;
+  }
+
+  // returnWorkBook: true -> raw workbook class
```

Comment:
Can you clarify why you would set this value one way or another?

Reply:
in our usecase, we've finished processing and just want to print the file out so we could simplify things; however, i'm taking the view that it's possible someone else is a consumer of the hdf-converters library and they might have some usecase where they'd like to further edit the workbook.